### DISCUSSION TITLE
When I use 'await session.commit()', the field content of my instance has become an error

### DISCUSSION THREAD

**Discussion by HaleTime** [fastapi/sqlmodel#1258] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the SQLModel documentation, with the integrated search.
- [X] I already searched in Google "How to X in SQLModel" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to SQLModel but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to SQLModel but to [SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import uuid
from sqlalchemy.ext.asyncio import AsyncSession
from app.core.db import async_engine
from sqlmodel import SQLModel, Field


class Test(SQLModel, table=True):
    id: uuid.UUID = Field(default_factory=uuid.uuid4, primary_key=True)
    name: str = Field(max_length=255)


async def main():
    async with async_engine.begin() as conn:
        await conn.run_sync(SQLModel.metadata.create_all)

    async with AsyncSession(async_engine) as session:
        test = Test(name='test')
        session.add(test)
        await session.commit()
        await session.refresh(test)
        return test


if __name__ == '__main__':
    import asyncio

    asyncio.run(main())
```

### Description

When I use 'await session.commit()', the field content of my instance has become an error, like 
(sqlalchemy.exc.MissingGreenlet) greenlet_spawn has not been called; can't call await_only() here. Was IO attempted in an unexpected place?
[SQL: SELECT test.id AS test_id, test.name AS test_name 
FROM test 
WHERE test.id = %s]
[parameters: [{'pk_1': UUID('84f97a3a-107b-4ae6-81bb-f04295cf0d3f')}]]

### Operating System

Windows

### SQLModel Version

0.0.22

### Python Version

3.10

**Comment by monchin** [comment]
This is a sqlalchemy issue. Try [this](https://stackoverflow.com/questions/74252768/missinggreenlet-greenlet-spawn-has-not-been-called)

**Comment by YuriiMotov** [comment, marked as answer]
Your code works well

SQLAlchemy marks object as expired after calling `commit`.
Next time you are attempting to use them it will try to refresh that object.
But it will fail since you are using async connection and all calls should be awaited.

So, after `commit` you should `refresh` objects to use them later
